### DISCUSSION TITLE
HW acceleration

### DISCUSSION THREAD

**Discussion by GMagician** [blakeblackshear/frigate#14406] [Ask A Question] [answered]
Where do I can see benefit between having:
ffmpeg:
  **hwaccel_args: preset-rk-h264**
  input_args: preset-rtsp-restream-low-latency
  output_args:
    record: preset-record-generic-audio-copy
or not?

theoretically I should see a difference in cpu usage since video decoding should be done by GPU.

CPU usage doesn't change, neither in go2rtc nor in other "cpu metrics"

go2rtc has #video=copy#audio=copy maybe because of no decoding since stream is just copied?

Thanks

**Comment by NickM-27** [comment]
go2rtc cpu usage will not change, the ffmpeg cpu usage in camera metrics tab should be lower

**Reply by GMagician** [reply]
I can't see any evaluable change in %
  with HW acceleration camera 1=4.2% and camera2=3.9%
  without HW acceleration camera 1=4.5% and camera2=5.9%
but values are continuously changing (up to 8% both modes)

**Reply by NickM-27** [reply]
what resolution stream are you using for detect?

**Reply by GMagician** [reply]
sub channel 720x480 can't go lower

**Reply by NickM-27** [reply, marked as answer]
right, that is pretty low, so the difference between hardware and software decoding will not be that much compared to if a larger stream was being decoded

**Reply by GMagician** [reply]
thanks for clarifications
